elements/src/main/java/com/manydesigns/elements/crypto/KeyManager.java

```diff
@@ -170,8 +177,16 @@ private String getPassPhrase(String passphrasePath) throws IOException, InvalidP
                 passPhrase.append(line);
             }
         } catch (IOException e) {
-            logger.error("getPassPhrase: " + e.getMessage(), e);
-            throw e;
+            logger.warn("getPassPhrase: " + e.getMessage());
+            logger.info("trying to get passphrase from env variables....");
+
+            String passPhraseEnv = System.getenv("PORTOFINO_PASSPHRASE");
+            if (passPhraseEnv == null) {
+                logger.error("passphrase not found! exiting...");
+                System.exit(1);
```

Comment:
System.exit è una cattiva idea in una libreria che va in esecuzione in un server con potenzialmente altre applicazioni in esecuzione, o anche solo altri processi della stessa applicazione. Basta lanciare l'eccezione e assicurarsi che il chiamante la gestisca.

Reply:
La system.exit l'ho usata come "strada veloce" perchè vorrei che se la passphrase è richiesta e non esiste, portofino interrompa l'avvio invece di continuare e poi lanciare eccezioni a runtime di KeyManager non inizializzato

Reply:
Ma per fare questo puoi lanciare eccezione come faceva prima, e non catturarla in PortofinoSpringConfiguration dove il keymanager viene inizializzato. In questo modo, il contesto di Spring non si inizializza e l'applicazione non parte.

Reply:
In realtà guardando il codice `PortofinoSpringConfiguration:89` ho visto che c'è un catch che nasconde questa cosa e fa partire lo stesso l'app...
come vogliamo procede? sicuramente il catch è da togliere

Reply:
Io lo toglierei proprio il catch. Credo che ci sia perché l'ho copiato tal quale da Portofino 4. Questa parte sulla crittografia non l'ho scritta io e non la conosco.